### DISCUSSION TITLE
Ending an OpenOCD process at the end of debugging

### DISCUSSION THREAD

**Discussion by DavenportEmma** [eclipse-cdt-cloud/cdt-cloud#27] [Q&A]
Hi,

I'm using Theia Blueprint with the cdt-gdb-vscode extension to debug an application on the Polarfire SoC. The debug configuration can be seen below. An OpenOCD instance is started and gdb connects to it. I am able to step through code and debug as expected, however when ending the debug session the instance of OpenOCD stays active and needs to be killed manually. Is there any way to configure the debug session to kill the OpenOCD process when the deubg session ends? Thanks!

```    
        "configurations": [{
        "name":"debug-smp",
        "type":"gdbtarget",
        "request":"launch",
        "program": "path/to/executable",
        "gdb": "path/to/gdb",
        "logFile": "path/to/logfile",
        "openGdbConsole": true,
        "initCommands": [
            "set mem inaccessible-by-default off",
            "set $target_riscv=1",
            "set architecture riscv:rv64"
        ],
        "preRunCommands": [
            "thread apply all set $pc=_start",
            "thread apply all set $mie=0",
            "thread apply all set $mip=0",
            "thread apply all set $mstatus=0"
        ],
        "imageAndSymbols": {
            "symbolFileName": "path/to/executable",
            "imageFileName": "path/to/executable"
        },
        "target": {
            "host": "localhost",
            "port": "3333",
            "server": "path/to/openocd",
            "serverParameters": ["--command", "set DEVICE MPFS", "--file", "board/microsemi-riscv.cfg", "--command", "set COREID 1"],
            "serverStartupDelay": 5000,
            "cwd": "${workspaceFolder}"
        }
    }]
    }
```

**Comment by planger** [comment]
@jonahgraham Probably you are the best person to answer this question? Thanks in advance!

**Comment by jonahgraham** [comment]
Thanks @ConorDavenport for raising this. This looks like a shortcoming of cdt-gdb-adapter, I have made an issue from this discussion in https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/issues/253
